lib/Sema/CSApply.cpp

```diff
@@ -6550,7 +6550,19 @@ maybeDiagnoseUnsupportedDifferentiableConversion(ConstraintSystem &cs,
           maybeDiagnoseFunctionRef(unwrappedFnExpr);
           return;
         }
+      } else if (auto conv = dyn_cast<FunctionConversionExpr>(semanticExpr)) {
```

Comment:
There is also `ABISafeConversionExpr` which does the same thing for l-value types, in case that matters in some other case.